services/ai/src/main/java/demo/mesharena/ai/AI.java

```diff
@@ -99,23 +120,52 @@ private void checkArenaInfo() {
             int defendZoneBottom = obj.getInteger("defendZoneBottom");
             int defendZoneLeft = obj.getInteger("defendZoneLeft");
             int defendZoneRight = obj.getInteger("defendZoneRight");
-            arenaInfo = new ArenaInfo(defendZoneTop, defendZoneLeft, defendZoneBottom, defendZoneRight, new Point(goalX, goalY));
+            int scoreA = obj.getInteger("scoreA");
+            int scoreB = obj.getInteger("scoreB");
+
+            if (arenaInfo == null || (arenaInfo.scoreA != scoreA || arenaInfo.scoreA != scoreB)) {
+              // the goals changed - new game started
+              try (Scope startGame = tracer.buildSpan("new_game")
```

Comment:
"new game" is a bit misleading as it happens after any new goal

Reply:
If I understand correctly, what we're doing here is create a span that will live for as long as there isn't a new goal scored? I guess this isn't really a typical case of tracing, or is it? (I mean, it seems quite particular due to the never-ending game loop)

Reply:
What do you think if we set a constant lifetime for this span instead of relying on goals, such as 1 or 5 or 10 seconds for instance?

Reply:
In this context the new game is when the score changes. The idea is to have one trace per goal and player - this way we can trace players behavior until somebody scores. How we model it is totally up to us, this made behavior made sense for me. I would be against finishing the span on timer base.

The never ending game would have an infinite number of spans. I think it's correct to model it this way, but might be overwhelming for the tracing backend.

Reply:
But it also means that if no goal is scored for a long time, which can happen in some scenario (ex: when we introduce a second ball then players are unable to chose a direction), then the span will grow infinitely. Is it a problem or not on the backend side?

Reply:
On large number of spans the in memory storage (the dafault in istio) should be dropping spans or OOM.

In the UI you will see that number of spans is increasing so indicating a running transaction.